server/src/main/java/keywhiz/inject/InjectorFactory.java

```diff
@@ -0,0 +1,25 @@
+package keywhiz.inject;
+
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import io.dropwizard.db.ManagedDataSource;
+import io.dropwizard.setup.Environment;
+import keywhiz.KeywhizConfig;
+import keywhiz.ServiceDataSourceModule;
+import keywhiz.ServiceModule;
+
+public final class InjectorFactory {
+  private InjectorFactory() {}
+
+  public static Injector createInjector(KeywhizConfig config, Environment environment) {
+    return Guice.createInjector(
+        new ServiceDataSourceModule(),
```

Comment:
It just means that the module will get references to dependencies "indirectly" via Guice. Modules are a lot less magical than they seem.

So, let's say you have some provider method in your module:

```
@Provides Bar bar(Foo foo) {
  return new Bar(foo);
}
```

Guice will locate the binding (instructions on how to create an instance) for Foo, spine one up, and pass it into this method as an argument. A different but mostly-equivalent style would be:

```
Module MyModule extends AbstractModule {
  private final Foo foo;

  public MyModule(Foo foo) {
    this.foo = foo;
  }

  @Provides Bar bar() {
    return new Bar(foo);
  }
}
```

So, instead of having Guice create the Foo instance on your behalf, you're directly supplying an instance yourself. The downside here is that the Foo is just a field, so it's effectively a singleton. If you want some other pattern (every new Bar also gets a new Foo), the first style is more flexible because it leaves that decision to someone else.

Generally, I would always recommend the first approach and reserve passing in constructor arguments for things which cannot be created by Guice (in our case, DropWizard-ish configuration), usually because they're instantiated by a framework before the Injector is ever created.